active/0002-centalized-mgmt-resource-types-provider-versions.md

```diff
@@ -0,0 +1,262 @@
+
+### Proposed Changes
+
+- [Updating `bicepconfig.json` merge semantics](#updating-bicepconfigjson-merge-semantics)
+- [Resolution of namespace identifiers by VSCode extension when inspecting published module sources](#resolution-of-namespace-identifiers-by-vscode-extension-when-inspecting-published-module-sources)
+- [Wildcard semantics for provider versions in `bicepconfig.json`](#wildcard-semantics-for-provider-versions-in-bicepconfigjson)
+
+## Detailed design
+
+### Client side changes
+
+#### Add a new syntax for the provider import declaration statements: `import {providerName} [as {optionalAlias}]`
+
+The example below shows a usage of the proposed syntax
+
+```bicep
+provider az
+provider kubernetes as k8s // uses optional aliasing of the
+```
+
+The resolution of the symbols will be resolved from the builtin `bicepconfig.json` under a new section:
+
+```json
+{
+  // prior bicepconfig.json sections
+  // ...
+
+  "providers": {
+    "az": {
+      // A package reference string (below) gestures Bicep to source the types from the registry address.
+      "source": {
+        "registry": "mcr.microsoft.com/bicep/providers",
+        "version": "0.2.3",
+      },
+      "isDefaultImport": true // Gestures Bicep to implicitly import the namespace identifier to the global scope
+    },
+    "kubernetes": {
+      // Using the value "builtin" (below) gestures Bicep to source types from the statically compiled NuGet package reference
+      "source": {
+        "builtin": true
+      },
+      "isDefaultImport": false
+    }
+  }
+}
+```
+
+Since previous provider declaration syntax forms continue to be supported, we introduce the following constraints:
+
+- To ensure consistency with pre-existing handling of `bicepconfig.json` the configuration file closest to the Bicep file in the directory hierarchy is used.
+- The `sys` namespace is coupled to the Bicep bits version and cannot be overriden or dynamically uploaded using the mechanism described above
+
+#### Remove support for syntax `import '{providerName}@{providerVersion}' [as {optionalAlias}]`
+
+The syntax below currently in use to import built-in providers will become invalid and result in a diagnostic as a result of this change
+
+```bicep
+provider 'sys@1.0.0'
+provider 'kubernetes@1.0.0' as k8s
+```
+
+#### Change the delimiter '@' used by the supported provider declaration syntax to ':' to align with module reference syntax and increase consistency
+
+The syntax shown below uses ':' instead of '@' as a delimiter for provider declaration statements
+
+```bicep
+provider 'br:mcr.microsoft.com/bicep/providers/az:0.2.3' // notice its NOT using az@0.2.3
+provider 'br/public:az:0.2.3' as myAz
+```
+
+## Drawbacks
+
+- Costumers may be confused by the config merge semantics behavior
+- Costumers will not be able to tell from inspecting sources alone what is the provider version used to resolve resource types
+
+## Alternatives
```

Comment:
```suggestion
## Alternatives considered
```

[Very minor, totally optional]: "Alternatives" to me implies that what is listed below is still a viable option. "Alternatives considered," however, implies that you thought the following through but decided not to do it. There's an implied "and rejected" following "Alternatives considered"

Reply:
Yeah I think that works, however I think likely this should be a change made in the template. The REP process delineated is to follow the template and this is what I did. I think your thoughts here are valid and should be discussed as a separate issue on the REP process